service/src/main/kotlin/app/cash/backfila/api/CreateAndStartBackfillAction.kt

```diff
@@ -0,0 +1,41 @@
+package app.cash.backfila.api
+
+import app.cash.backfila.BackfillCreator
+import app.cash.backfila.dashboard.BackfillStateToggler
+import app.cash.backfila.protos.service.CreateAndStartBackfillRequest
+import app.cash.backfila.protos.service.CreateAndStartBackfillResponse
+import app.cash.backfila.service.persistence.BackfillState
+import misk.MiskCaller
+import misk.scope.ActionScoped
+import misk.security.authz.Unauthenticated
+import misk.web.Post
+import misk.web.RequestBody
+import misk.web.RequestContentType
+import misk.web.ResponseContentType
+import misk.web.actions.WebAction
+import misk.web.mediatype.MediaTypes
+import javax.inject.Inject
+
+class CreateAndStartBackfillAction @Inject constructor(
+  private val caller: @JvmSuppressWildcards ActionScoped<MiskCaller?>,
+  private val backfillCreator: BackfillCreator,
+  private val backfillStateToggler: BackfillStateToggler
+) : WebAction {
+  @Post("/create-and-start-backfill")
+  @RequestContentType(MediaTypes.APPLICATION_PROTOBUF)
+  @ResponseContentType(MediaTypes.APPLICATION_PROTOBUF)
+  // TODO authenticate but any service
+  @Unauthenticated
+  fun createAndStartBackfill(
+      @RequestBody request: CreateAndStartBackfillRequest
+  ): CreateAndStartBackfillResponse {
+    val caller = caller.get()!!
+    val service = caller.service!!
+
+    val id = backfillCreator.create(service, service, request.create_request)
+
+    backfillStateToggler.toggleRunningState(id.id, caller, BackfillState.RUNNING)
```

Comment:
What should the end state of it created but didn't start running be?

Reply:
normally it's created PAUSED

Reply:
Yeah but if the start fails is creating a paused backfill ok? Or should it clean up after itself?

Reply:
yea, seems harmless to create one that never runs